SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1172,11 +1163,33 @@ void USpatialActorChannel::UpdateSpatialPosition()
 			return;
 		}
 	}
+	
+	// Check that the Actor has satisfied both minimum thresholds OR either of the maximum thresholds
+	bool bSatisifiedPositionUpdateRequirements = false;
+
+	const float TimeSinceLastPositionUpdate = NetDriver->GetElapsedTime() - TimeWhenPositionLastUpdated;
 
-	// Check that the Actor has moved sufficiently far to be updated
-	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
 	FVector ActorSpatialPosition = SpatialGDK::GetActorSpatialPosition(Actor);
-	if (FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate) < SpatialPositionThresholdSquared)
+	const float DistanceTravelledSinceLastUpdateSquared = FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate);
+
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+	const float SpatialMinimumPositionThresholdSquared = FMath::Square(SpatialGDKSettings->PositionUpdateThresholdMinDistance);
```

Comment:
Unless we expect to change these are runtime it might be better to cache the results somewhere (as we are having to fetch them from somewhere else in memory anyway)

Reply:
I think we might still want to allow users to change this at runtime? However, if you believe this is not the case I will change it and cache them somewhere.

Reply:
In which case we might want the settings to specify a square distance threshold but I'm not sure it matters that much until we profile.